### ISSUE TITLE
Armored Core 3: Z-Fighting with every renderer

### ISSUE THREAD

**Issue by Helium-4** [PCSX2/pcsx2#2398]
***PCSX2 version:***
  v1.5.0-dev-2334-gefc1274b2

***PCSX2 options:***
 Default

***Plugins used:***
 Default

***Description of the issue:***
Some surface will exhibit some Z-fighting at specific angles, it is present in every renderer, it's sometimes very subtle and sometimes completely impossible to miss, here's an example of the latter
![gsdx_20180420214844](https://user-images.githubusercontent.com/30066566/39070712-b50ccec0-44e4-11e8-942b-4d29ba081111.png)
[Armored core 3 Zfighting.zip](https://github.com/PCSX2/pcsx2/files/1933487/Armored.core.3.Zfighting.zip)

Setting the EE/FPU Rounding mode to Positive or Nearest fixes that issue, although setting it on Positive creates other gameplay affecting issues as well as seams in some terrain textures, I didn't have time to thoroughly test it on Nearest but so far I did not run into anything noticeable while replaying most of the missions.

***How to reproduce the issue:***
1.Start any mission in a level environment that exhibits those issues

***Last known version to work:***
 Never

***PC specifications:***
  Intel i7 4770K
  Nvidia GTX 1070 (390.77)
  Windows 7.

**Comment by refractionpcsx2** [comment]
If you could let us know what the result is with nearest, if that's OK we might be able to just add that to the game database so it automatically turns on.

**Comment by Helium-4** [comment]
Can confirm that using Nearest doesn't cause any issue and fixes the problem.

**Comment by refractionpcsx2** [comment]
Thanks for letting us know

**Comment by Helium-4** [comment]
Same issue also affects Silent Line: Armored Core,, seems the same fixes also applies.
I'll do a playthrough with it on and report back.

**Comment by refractionpcsx2** [comment]
Yeah if you could post the results in the appropriate issue you've created, makes it easier to trace.

**Comment by TransparentBlue** [comment]
Not sure it's worth reopening  but Nearest does still cause a few issues of it's own which are worth documenting.
The intro cutscene of the "Disable Pulse generator" mission will have flicker where the 3D scene in the background is flipped diagonally for a frame for the duration of the cutscene, it is the only scene where this is an issue, other rounding settings do not have that specific issue.
Another more problematic issue you can go out of mission bounds if you use some hover type legs and go over water in the "Destroy massive weapon mission", though I've seen it happen on real hardware as well but it is rather rare whereas here it's somewhat more common (think 1 in 20 instead of 1 in 100),  using Negative or Chop / Zero prevents that specific issue from happening at all, Positive rounding will make it happen 100%.

**Comment by refractionpcsx2** [comment]
Okay, which one is the best option then? If a different rounding works better and causes less problems, we can change it

**Comment by TransparentBlue** [comment]
I'd say nearest is still fine, because it merely has two noticeable issues instead of a bunch.
Maybe once #5188 or something like RPCS3 accurate float happens that could be fully fixed without drawbacks.

**Comment by refractionpcsx2** [comment]
Reopening since the previous fix broke some stuff in the game.  Though it's possible this isn't fixable in hardware mode, but software should be fine.

**Comment by TransparentBlue** [comment]
Software mode has the same exact issue.

**Comment by TransparentBlue** [comment]
Same situation as GT4 more or less, either break one thing or break another, though technically there's also positive rounding but that breaks another thing entirely.

**Comment by JordanTheToaster** [comment]
Fixed by #12001 with VU0 Add/Sub Mul/Div.
